src/wp-includes/class-wp-query.php

```diff
@@ -1344,6 +1344,38 @@ public function parse_tax_query( &$q ) {
 		do_action( 'parse_tax_query', $this );
 	}
 
+
+	/**
+	 * Load comments from cache where possible, else make query and cache.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param string $comments_request SQL query.
+	 * @return WP_Comment[] Array of `WP_Comment` objects.
+	 */
+	protected function get_comments( $comments_request ) {
+		global $wpdb;
+
+		$key          = md5( $comments_request );
+		$last_changed = wp_cache_get_last_changed( 'comment' );
+		if ( str_contains( $comments_request, $wpdb->posts ) ) {
+			$last_changed .= wp_cache_get_last_changed( 'posts' );
```

Comment:
```suggestion
			$last_changed .= ':' . wp_cache_get_last_changed( 'posts' );
```

Is there a practical limit to the key length?

Reply:
Fixed